digdag-core/src/main/java/io/digdag/core/agent/MultiThreadAgent.java

```diff
@@ -98,12 +118,18 @@ public void run()
                                     logger.error("Uncaught exception. Task queue will detect this failure and this task will be retried later.", t);
                                     errorReporter.reportUncaughtError(t);
                                 }
+                                finally {
+                                    synchronized (taskCountLock) {
+                                        activeTaskCount--;
```

Comment:
With `SynchronousQueue`, calling `executor.submit()` can be blocked holding `taskCountLock`. As a result, a thread that tries to decrement the counter here can be potentially blocked forever?